tests/parsing/LineParserTest.py

```diff
@@ -14,13 +14,19 @@ def test_empty_line(self):
         self.check_data_set('\n \n \n')
 
     def test_comment_parsing(self):
+        logger = logging.getLogger()
         self.check_data_set('# comment only$§\n',
                             output_comment='# comment only$§')
         self.check_data_set('   ; comment only  \n',
                             output_comment='; comment only')
         self.check_data_set('   ; \\comment only  \n',
                             output_comment='; comment only')
         self.check_data_set('#', output_comment='#')
+        with self.assertLogs(logger, 'WARNING') as warn:
+            self.check_data_set('##\n', output_comment='##')
+        self.assertEqual(warn.output[0][13:], 'The comment does not have ' +
+                                              'whitespace before or after # ' +
+                                              'in: ##')
```

Comment:
Tests are also not sufficient to detect the cases I mentioned and preferably use  this
```
 self.assertEqual(len(warn.output), 1)
self.assertIn(warn.output[0], 'WARNING:root:The comment does not have whitespace before or after # in: ##')
```